### PR TITLE
Increment MultiOptimizer iterations after each apply_gradients

### PR DESCRIPTION
**Pull request by lucasdavid**
MultiOptimizer does not call `super().apply_gradients`, and therefore does not update its `iterations: tf.Variable<type=int, shape=[]>` as the other optimizers in Tensorflow.
This leads to buggy behavior when training with a more complex `train_step` that relies on this variable for scheduling *learning rate* or other hyper-parameters such as *increasing/decreasing weights* for individual losses.

### CODE REVIEW

**Comment by bhack** [comment]
Do you have a test to cover this use case?

**Comment by lucasdavid** [comment]
I can write it. Do I post it in the comments or as a new unit test?

Unrelated to my problem: the testcase `tensorflow_addons/optimizers/tests/discriminative_layer_training_test:test_model` does not have any checkings, and is just a sanity test.

**Comment by lucasdavid** [comment]
The failing check is telling me `using tf.control_dependencies() is not allowed in the codebase.`

I'm not sure if removing it in this scenario is a good idea though.

**Comment by lucasdavid** [comment]
Another way to fix this (and maybe cleaner) would be to just defer to the inner optimizer's iterations:
```
class MultiOptimizer(...):
  @property
  def iterations(self):
    return self.specs[0]['optimizer'].iterations
```

**Comment by AakashKumarNain** [comment]
@bhack I will look into this today or tomorrow if I get some time

**Comment by fsx950223** [comment]
Add the file to the list https://github.com/tensorflow/addons/blob/ee5798c189f3903463b21f0d7189f480c47c365a/tools/testing/source_code_test.py#L177

**Comment by lucasdavid** [comment]
@fsx950223 done. All checks pass now. Thank you. :-)

**Review by fsx950223 — APPROVED** [review]
